x/dex/types/message_multi_hop_swap.go

```diff
@@ -70,17 +70,32 @@ func (msg *MsgMultiHopSwap) Validate() error {
 	if len(msg.Routes) == 0 {
 		return ErrMissingMultihopRoute
 	}
-
+	// set the expected tokenIn as the first token in the first route.
+	// if the route is empty, don't set tokenIn and let route length check error.
+	var tokenIn string
+	if len(msg.Routes[0].Hops) != 0 {
```

Comment:
We are already checking that len(msg.Routes[i].hopes) >= 2 below. I would refactor so we don't have to do this check twice. Also maybe better to keep this parallel with the `expectedExitToken` code below. May have to refactor both

Reply:
agree this function is getting ugly. will refactor